labs/b5.md

```diff
@@ -0,0 +1,364 @@
+
+Moreover, compared to TCP every single UDP send requires a UDP receive (recv) per datagram. While for TCP you pass a stream of data that is transparently split into some number of sends and the data stream can be reconstructed with a single recv call.
+
+## Ports
+[port list]: https://www.iana.org/assignments/service-names-port-numbers/service-names-port-numbers.xhtml
+
+Ports define a service endpoint, broadly speaking -- ports mark a point of traffic ingress and egress.  Ports are represented by a 16 bit number meaning thus ranging from 0 to 65535.  Ports from 0 to 1023 are well known ports, i.e. system ports.  Using these ports usually has a stricter requirement.  1024 to 49151 are registered ports. IANA maintains the official [list][port list] of well-known and registered ranges. The remaining ports from 49152 to 65535 are ephemeral ports which can be dynamically allocated for communication sessions on a per request basis.
+
+Some port numbers for well known services are as follows:
+Service     | Port
+------------|------
+HTTP        |80
+HTTPS       |443
+
+
+# Sysadmin Commands
+As a sysadmin, trying to diagnose network issues can often be pretty challenging. Given the scale and complexity of networks, it's tough trying to narrow down the scope of a problem to a point of failure. What follows is a list of commands/tools that can help with triaging problems. There are a lot of tools and we don't expect you to memorize every single detail. However, it is important to know what tools exist and when to use them when a problems inevitably arise. If you ever need more details the `man` pages for these commands are a great place to turn to for reference.
+
+Tools also tend to overlap in functionality -- for example there are multiple tools that can display interface information or test connectivity.  When possible, it is a good idea to use multiple tools to cross-check one another.
+
+Note that when it comes to real world networks there are even more factors to consider that we haven't touched on like network security. For example, two machines can have a fully functioning connection but if one machine has been configured to drop all packets then it might seem as if they aren't connected. 
+
+So take the output of these tools with a grain of salt, they a means of narrowing down issues. It is important not to misinterpret outputs or jump to conclusions too quickly.
+
+1. `hostname`
+A simple and straightforward command that can display information about a host, IP addresses, FQDN, and etc.
+
+2. `ping`
+ Another simple command, most of the time you'll be using ping as a first step towards testing connectivity. If ping can't reach a host then there is likely an issue with connectivity.
```

Comment:
what does `ping` do to test connectivity? (i.e. tell them about ICMP)

Reply:
I don't think I should go too deep into the protocol.  I'll make a broad mention that it essentially sends packets to the destination expecting a response and link for additional information.